frontend/src/metabase/common/components/QueryColumnPicker/QueryColumnPicker.tsx

```diff
@@ -149,33 +166,54 @@ export function QueryColumnPicker({
       color,
       checkIsColumnSelected,
       handleSelect,
+      withInfoIcons,
     ],
   );
 
+  const renderItemExtra = useCallback(
+    item => (
+      <QueryColumnInfoIcon
+        query={query}
+        stageIndex={stageIndex}
+        column={item.column}
+        position="top-start"
```

Comment:
Design comment:

It's a bit annoying that this popover is shown on top of other icons that can trigger it. If I'm going up the list with my mouse cursor, the popover will prevent me from inspecting other info icons.

https://github.com/metabase/metabase/assets/6830683/443f087f-89c3-48bc-892f-bc4cc6325ff8

Can we just show the popover on the right, so it does not cover anything, [like this](https://github.com/metabase/metabase/assets/6830683/0c30a32e-0562-419b-a9c0-a9293c774520)?

cc: @mazameli

Reply:
Yeah the tricky thing is that, on small screens if the popover does not fit on the right of the icon, it will be flipped to the left and overlap on top of the item which is not what we want. 

I'll try to update the default positioning logic to avoid this.

Reply:
To the right is ideal, but above is second-best, given that I think it's a reasonable assumption that slightly more of the time if you're looking at the descriptions of multiple columns, you'd start from the top and work your way down.